src/RemoteAPI/class-remote-api-base.php

```diff
@@ -151,14 +151,24 @@ public function get_api_url( array $query ): string {
 	 */
 	public function get_items( $query, $associative = false ) {
 		$full_api_url = $this->get_api_url( $query );
+		$options      = $this->set_request_options();
+		$response     = wp_safe_remote_get( $full_api_url, $options );
 
-		$result = wp_safe_remote_get( $full_api_url, array() );
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+		if ( 200 !== $response_code ) {
```

Comment:
Any particular reason of adding this code? Does API returns anything else other than 200 which is not a 4xx or 5xx?

Reply:
> Do you see any particular issue with the code? I'm unsure I understand the reason for this question.

As far as I know `is_wp_error` already handles everything which is not a 2xx.

Just for an example say API returns 204/206 then we will throw the error as per this code which is technically not right.

Reply:
Trying to remember why this code was introduced, it was probably because an experimental API endpoint had a particular behavior. But this is not applicable right now, so removed it.